Modules/quote/Controller/QuotelistController.php

```diff
@@ -224,14 +242,55 @@ public function editnewuserAction($id_space, $id) {
         if ($tableHtml != "") {
             $form->setTitle(QuoteTranslator::Description($lang));
         }
-
+        $modelClient = new ClClient();
+        $clients = $modelClient->getAll($id_space);
+        $clientIds = ["0"];
+        $clientNames = ["-- --"];
+        forEach($clients as $client) {
+            array_push($clientIds, $client['id']);
+            array_push($clientNames, $client['name']);
+        }
+        // set selected client by default
+        $selectedClientId = $info['id_client'] ?? $clientIds[0];
+        $addressToDisplay = "";
+        if ($info['address'] && $info['address'] != "") {
+            $addressToDisplay = $info['address'];
+        } else {
+            $addressToDisplay = $modelClient->getAddressInvoice($id_space, $selectedClientId) ?? "";
+        }
+        $form->addHidden('id_space', $id_space);
         $form->addText("recipient", QuoteTranslator::Recipient($lang), true, $info['recipient']);
-        $form->addTextArea("address", QuoteTranslator::Address($lang), true, $info['address']);
-
+        $form->addEmail("recipient_email", QuoteTranslator::Recipient($lang) . " " . CoreTranslator::Email($lang), false, $info['recipient_email']);
+        $form->addTextArea("address", ClientsTranslator::Client($lang) . " " . QuoteTranslator::Address($lang), true, $addressToDisplay);
+        $form->addSelect('id_client', ClientsTranslator::Client($lang), $clientNames, $clientIds, $selectedClientId);
+
+        $modelClientPricing = new ClPricing();
+        $pricings = $modelClientPricing->getPricingByClient($id_space, $selectedClientId);
+        $pricingIds = [];
+        $pricingNames = [];
+        forEach($pricings as $pricing) {
+            array_push($pricingIds, $pricing['id']);
+            array_push($pricingNames, $pricing['name']);
+        }
+        
+        if ($id > 0) {
+            // if quote has a client: get this client's pricing
+            if ($info['id_client'] && $info['id_client'] != 0) {
+                // (A client has 0 to 1 pricing)
+                $selectedPricing = $modelClientPricing->getPricingByClient($id_space, $info['id_client'])[0]['id'];
+                // if quote has a pricing: get it
+            } else if ($info['id_belonging'] && $info['id_belonging'] != 0 ) {
+                $modelPricing = new ClPricing();
+                $pricing = $modelPricing->get($id_space, $info['id_belonging']);
```

Comment:
looking at code, id_belonging, in code, refer to a client not a clpricing, in quote.getAllInfo:

            $data["recipient"] = $modelUser->getUserFUllName($data["id_user"]);
            $resps = $modelUserClient->getUserClientAccounts($data["id_user"], $id_space);
            if ($resps && !empty($resps)) {
                $data["address"] = $modelClient->getAddressInvoice($id_space, $resps[0]["id"]);
                $data["id_belonging"] = $resps[0]["id"];

Reply:
in db, old quotes references to id_belongings match with cl_pricings ids...

Reply:
example :  quote 47
=> id_belonging = 23
=> id_space = 13
cl_pricing 23 id_space = 13
cl_client 23 is not linked to space 13

Reply:
i'm really missing something there